### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by kukabi**
Boilerplate, some UI utilities. Import the wallet library and make a sample call through JNI @ the main activity.

### CODE REVIEW

File: app/src/main/cpp/native-lib.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * LOGI("Name is %s", name);
+ */
+#define LOGE(...) __android_log_print(ANDROID_LOG_ERROR,    LOG_TAG, __VA_ARGS__)
+#define LOGW(...) __android_log_print(ANDROID_LOG_WARN,     LOG_TAG, __VA_ARGS__)
+#define LOGI(...) __android_log_print(ANDROID_LOG_INFO,     LOG_TAG, __VA_ARGS__)
+#define LOGD(...) __android_log_print(ANDROID_LOG_DEBUG,    LOG_TAG, __VA_ARGS__)
+
+extern "C"
+JNIEXPORT jstring JNICALL
+Java_com_tari_android_wallet_ui_activity_MainActivity_stringFromJNI(
```
Comment (StriderDM) [inline app/src/main/cpp/native-lib.cpp]: This method can be removed

File: app/src/main/cpp/native-lib.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        JNIEnv *env,
+        jobject /* this */) {
+    std::string hello = "Message from JNI!";
+    return env->NewStringUTF(hello.c_str());
+}
+
+extern "C"
+JNIEXPORT void JNICALL
+Java_com_tari_android_wallet_ui_activity_MainActivity_changeCountJNI(
```
Comment (StriderDM) [inline app/src/main/cpp/native-lib.cpp]: This method can be removed

File: app/src/main/cpp/native-lib.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    TariPrivateKey *pk = private_key_generate();
+    ByteVector *bytes = private_key_get_bytes(pk);
+    unsigned int bytes_length = byte_vector_get_length(bytes);
+    std::string s = std::string("");
+    for (int i = 0; i <= bytes_length; i++)
+    {
+        unsigned int byte = byte_vector_get_at(bytes,i);
+        s += static_cast<char>(trunc(byte/2)); //should be converted to hex, div by 2 to ensure utf8 valid
+    }
+    return pEnv->NewStringUTF(s.c_str());
```
Comment (StriderDM) [inline app/src/main/cpp/native-lib.cpp]: This was part of sample test code provided.

Note that *pk and *bytes go out of scope after this return however the memory they are pointing to is not freed leading to a leak.

File: app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package com.tari.android.wallet.ui.widget
+
+import android.content.Context
+import android.graphics.Typeface
+import java.util.*
+
+enum class CustomFont(val fileName: String) {
```
Comment (StriderDM) [inline app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: What is the license for these fonts?

Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: This piece of code & fonts I just included as an example of custom font use. We may delete after we replace them with the fonts we'll receive from the designer.

Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: I don't think we would use Proxima Nova (just used it to make the demo look better really), and Font Awesome offers a free tier I think.

File: app/src/main/cpp/native-lib.cpp (hunk 2 above)
Comment (kukabi) [reply app/src/main/cpp/native-lib.cpp]: Just included this and stringFromJNI for demo purposes, deleting them.

File: app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt (hunk 4 above)
Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: https://fontawesome.com/license/free license for FontAwesome 5. FontAwesome can be pretty useful, and here's a list of free fonts in version 5: https://fontawesome.com/icons?d=gallery&m=free

Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: I have also included FontAwesome 4 in the project, which has some other useful fonts, and that one is free to use for anything it seems: https://fontawesome.com/v4.7.0/license/

File: app/src/main/cpp/native-lib.cpp (hunk 3 above)
Comment (kukabi) [reply app/src/main/cpp/native-lib.cpp]: @StriderDM I guess I'll just add before return:
```cpp
byte_vector_destroy(bytes); 
private_key_destroy(pk);
```

File: app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt (hunk 4 above)
Comment (kukabi) [reply app/src/main/java/com/tari/android/wallet/ui/widget/CustomFont.kt]: Removing them all in favor of the app fonts from the iOS app.

File: app/src/main/cpp/native-lib.cpp (hunk 3 above)
Comment (StriderDM) [reply app/src/main/cpp/native-lib.cpp]: This will work

**Pull request merged**